doc/dirty-tracking.md

```diff
@@ -0,0 +1,44 @@
+
+If the provider has switched from storing wrapped to unwrapped documents or vice versa, the CFM will identify this and mark the document as "dirty" immediately after opening it. This handled by the `requiresConversion` method. 
+
+The client might need to modify a document right after opening. If this is automatic without user interaction, the client might **not** want these changes to make the document "dirty". If autosaving is disabled the "UNSAVED" badge will be shown even though the user hasn't changed anything themselves. If autosaving is enabled the user might see the "UNSAVED" badge flash and disappear. Additionally when the saving of document is an indication of student progress, this initial immediate saving will make it appear the user has changed the document when really they have not. Because it is the responsibility of the client to tell the CFM when the content is dirty these cases can be handled by just not marking the document dirty.
+
+## Autosave
+When autosave is enabled by passing the `autoSaveInterval` configuration option, the CFM will try to autosave on this interval. If the document is not "dirty" then the autosave will do nothing. So when autosave is used, it is important that the client calls `client.dirty()` otherwise the document will not be auto saved.
+
+When the autosave loop sees the document is dirty, it will request the content from the client using the `getContent` event and then save the result.
+
+## Shared doc info
+The CFM normally includes the shared doc info in the original document that is being shared. This is so the author of the original document can update the shared document when they re-open the original document. 
+
+Some providers allow the user to not include this shared doc info in the saved file. This is useful if you want to send a document to someone but you don't want them to be able update the shared document.
+
+The shared doc info is related to the dirty state of the document because the CFM needs to know to save the original document once this shared doc info has been added to it. When the document is shared or unshared the CFM will mark the document as dirty. This is one case where the CFM will actually mark the document dirty itself.
+
+This is further complicated by the unwrapped option in the CFM. When the client configures the CFM to store the content unwrapped, it is the responsibility of the client to store the shared doc info inside of its content. Clients using the unwrapped option do this by listening for the `sharedFile` event from the CFM. After updating the content a client should set the dirty state so the CFM knows it should save the client's updated content.
```

Comment:
I think v3 is not doing this at the moment. There should be a story for it in the CODAP v3 PT board if there isn't.

Reply:
Good point. Here is a story: https://www.pivotaltracker.com/story/show/188523684